bluebrain/repo-patches/packages/highfive/package.py

```diff
@@ -79,6 +82,5 @@ def cmake_args(self):
             "-DHIGHFIVE_EXAMPLES:Bool=" + str(self.spec.satisfies("@develop")),
             "-DHIGHFIVE_UNIT_TESTS:Bool=" + str(self.spec.satisfies("@develop")),
             "-DHIGHFIVE_TEST_SINGLE_INCLUDES:Bool=" + str(self.spec.satisfies("@develop")),
-            "-DHIGHFIVE_USE_INSTALL_DEPS:Bool=ON",
-            "-DHDF5_NO_FIND_PACKAGE_CONFIG_FILE:Bool=ON",
+            "-DHIGHFIVE_USE_INSTALL_DEPS:Bool=Off",
```

Comment:
Spack is built on the premise that packages should use the dependencies they have been installed with.

Reply:
If I switch to `"-DHIGHFIVE_USE_INSTALL_DEPS:Bool=On"` I can't install `libsonata` and `highfive` at the same time, e.g.
```
spack env create hi5-libsonata
spack env activate -p hi5-libsonata

spack add highfive
spack add libsonata@0.1.17

spack concretize -f && spack install --jobs JOBS
```
Fails with
```
     90     /gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_externals/install_gcc-12.2.0-skylake/hpe-mpi-2.25.hmpt-mgws7w/bin/mpicxx -fPIC -O2 -g -DNDEBUG -sh
            ared -Wl,-soname,libsonata.so.0.1 -o libsonata.so.0.1.17 CMakeFiles/sonata_shared.dir/src/common.cpp.o CMakeFiles/sonata_shared.dir/src/config.cpp
            .o CMakeFiles/sonata_shared.dir/src/edge_index.cpp.o CMakeFiles/sonata_shared.dir/src/edges.cpp.o CMakeFiles/sonata_shared.dir/src/hdf5_mutex.cpp.
            o CMakeFiles/sonata_shared.dir/src/node_sets.cpp.o CMakeFiles/sonata_shared.dir/src/nodes.cpp.o CMakeFiles/sonata_shared.dir/src/population.cpp.o 
            CMakeFiles/sonata_shared.dir/src/report_reader.cpp.o CMakeFiles/sonata_shared.dir/src/utils.cpp.o CMakeFiles/sonata_shared.dir/src/version.cpp.o  
            -Wl,-rpath,::: -lhdf5-shared -lpthread -lmpi
  >> 91     /gpfs/bbp.cscs.ch/ssd/apps/bsd/2023-02-23/stage_compilers/install_gcc-4.8.5-haswell/binutils-2.37-l5h4u2/bin/ld: cannot find -lhdf5-shared
  >> 92     collect2: error: ld returned 1 exit status
  >> 93     make[2]: *** [libsonata.so.0.1.17] Error 1
```